pkgs/by-name/libresoc/libresoc-pyelftools.nix

```
@@ -0,0 +1,13 @@
+{
+  python39Packages,
+  fetchgit,
+}:
+python39Packages.pyelftools.overrideAttrs (_: rec {
+  name = "libresoc-pyelftools";
+  version = "v0.30";
+  src = fetchgit {
+    url = "https://git.libre-soc.org/git/pyelftools.git";
+    rev = version;
+    sha256 = "sha256-A9etnN7G24/Gu8YlV/YDpxZV+TG2eVXGx2ZjVnA9ZD4=";
+  };
```

```suggestion
  version = "0.30";
  # libre-soc project uses a mirror at https://git.libre-soc.org/git/pyelftools.git, but no commits have diverged, so use upstream nixpkgs package of pyelftools
```

IIRC during our discussion you said that libre-soc had diverged from upstream pyelftools, but looking at the repo at v0.30 https://git.libre-soc.org/?p=pyelftools.git;a=shortlog;h=refs/tags/v0.30, it appears to be the same? I think further proof of that is that nix computes the same hash https://github.com/NixOS/nixpkgs/blob/8596068d46e39547f0de0e73b5752bad3e87474a/pkgs/development/python-modules/pyelftools/default.nix#L20

However, I do think it's still worth pinning the version here in case the pyelftools in nixpkgs gets upgraded? although then I think we need to pin the hash too?

---

I just tried `nix build .#libresoc.verilog` on https://github.com/ngi-nix/ngipkgs/pull/168/commits/4cf5667583880efb05a02cdbb954243afa5dd0ba (i.e. 1 commit back) with these changes and it seemed to work. I got the same verilog output minus nix store paths that were in comments.

However I believe `fetchFromGitHub` will ignore the `rev` if there's a `hash`, so "pinning" `version` here probably doesn't accomplish anything 🤔 , we should set our own hash here (even if it would currently be the same)

---

`pyelftools` isn't actually mirrored yet though (see https://github.com/Libre-SOC-mirrors/pyelftools/branches and note we don't have the `master` branch yet) because we'd have to [give `workflows` permission to the mirror action](https://github.com/Libre-SOC-mirrors/pyelftools/actions/runs/8873601607/job/24359697295)
```
 ! [remote rejected] master -> master (refusing to allow a GitHub App to create or update workflow `.github/workflows/ci.yml` without `workflows` permission)
 ```
 
 This is what led me to looking up where pyelftools is originally coming from. It did feel a bit weird to me to have a bunch of mirrors from upstream pyelftools in GitHub -> libresoc cgit mirror -> Libre-SOC-mirrors GitHub mirror, but if you still thinks it makes sense to get it from `Libre-SOC-mirrors` I can give `workflows` permission so it can successfully mirror the master branch

---

Yeah pyelftools has GitHub workflows configured which none of the other mirrors did. One other dependency that is mirrored by libresoc cgit with a GitHub workflow was cached-property, but I think you ended up using it from nixpkgs here, so I didn't bother trying to make a mirror for that too

---

I am deleting https://github.com/Libre-SOC-mirrors/pyelftools since it's now unused